src/number_format.js

```diff
@@ -671,6 +671,16 @@ class NumberFormat extends React.Component {
       );
     }
 
+    /** Check for "paste event" with, replace any allowed decimal separator with desired decimal separator */
+    if (!format && decimalScale !== 0 && allowedDecimalSeparators.some(separator => value.indexOf(separator) !== -1)) {
+      const separator = decimalScale === 0 ? '' : decimalSeparator
```

Comment:
In the previous line we have a constraint for `decimalScale` to not be 0 so `decimal === 0` seems like it will never be true.. Can we fix this?